### DISCUSSION TITLE
Getting multi-value query params inside getServerSideProps

### DISCUSSION THREAD

**Discussion by hirani89** [vercel/next.js#18037] [Help]
Hi,
I have a query string as follows `news?state[]=2&state[]=3&state[]=4&date-from=2020-01-09&date-to=2020-05-07`

How do I get the state values as there are multiple values assigned to it?

Also, how do I get the date params as they have `-` in the name of the param?

**Comment by ljosberinn** [comment]
you can iterate over `new URL('http://n.com/news?state[]=2&state[]=3&state[]=4&date-from=2020-01-09&date-to=2020-05-07').searchParams.forEach`
